From 98945364f9ac58a4554155d8ced16bfbef700245 Mon Sep 17 00:00:00 2001
From: Cheng Yueh <cyueh@chromium.org>
Date: Wed, 8 Jul 2020 13:13:40 +0800
Subject: [PATCH] alsaucm: Fix ending with quotes commands

Upstream status:
https://lore.kernel.org/alsa-devel/20200709062412.348-1-cyueh@chromium.org/T/

If we use 'alsaucm -n -b -' with 'get "CapturePCM/Internal Mic"' then
the alsaucm will report error and stop immediately. The reason is that
the parse_line in usecase.c appends an empty argument if a command ends
with quotes.

This change adds a patch to fix the parse_line function in usecase.c.

Signed-off-by: Cheng Yueh <cyueh@chromium.org>
---
 alsaucm/usecase.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/alsaucm/usecase.c b/alsaucm/usecase.c
index 1ed91ea..d39a159 100644
--- a/alsaucm/usecase.c
+++ b/alsaucm/usecase.c
@@ -130,6 +130,8 @@ static int parse_line(struct context *context, char *line)
 							*line == '\n'))
 			line++;
 		c = *line;
+		if (c == '\0')
+			return 0;
 		if (c == '\"' || c == '\'') {
 			start = ++line;
 			while (*line && *line != c)
-- 
2.26.2

